server/stt/stt_newsroom_ninjs_formatter.py

```diff
@@ -36,10 +37,65 @@ def update_ninjs_stt_sources(self, ninjs):
 
             ninjs["source"] = "-".join(sorted_sources)
         except Exception as e:
-            logger.error(f"Error occurred when updating ninjs stt sources: {str(e)}")
+            logger.error(f"Error occurred when updating stt sources: {str(e)}")
+
+    def update_place(self, ninjs):
+        locators = [place.get("code") for place in ninjs.get("place", [])]
+        if locators:
+            ninjs["locators"] = locators
+
+    def update_body_from_content_profiles(self, ninjs):
+        extra = ninjs.get("extra", {})
+        profile = extra.get("content_profile_name", ninjs.get("profile"))
+        if not profile:
+            return
+        if profile.lower() in ("viiva", "sms"):
+            headline = ninjs.get("headline", "")
+            if headline:
+                body = ninjs.get("body_html", "")
+                ninjs["body_html"] = f"<p>{headline}</p>" + (body or "")
+
+    def update_subheadline(self, ninjs):
+        extra = ninjs.get("extra", {})
+        sttsubheadline = extra.get("sttsubheadline")
+        profile = extra.get("content_profile_name", ninjs.get("profile"))
+        if not sttsubheadline:
+            return
+        if profile.lower() in ("pikaplus"):
+            text = re.sub(r"<[^>]+>", "", sttsubheadline).strip()
+            if not text:
+                text = sttsubheadline.strip()
+            header = f"<h2>{text}</h2>"
+            body = ninjs.get("body_html", "")
+            ninjs["body_html"] = header + (body or "")
+
+    def update_sttversion(self, ninjs):
+        extra = ninjs.get("extra", {})
+        profile = extra.get("content_profile_name", ninjs.get("profile"))
+        if not profile:
+            return
+        versions = str(profile).strip()
+        ninjs["sttversion"] = versions
```

Comment:
There's an implementation note on the ticket, to use the [sttversion CV](https://github.com/superdesk/superdesk-stt/blob/develop/server/data/vocabularies.json#L1095) to get the `content_profile_name` to be used here.
```
Implementation note:
The CV was extended with content_profile_name.
It would be nice to use it in case we will need to add more content profiles in future
```

Therefor we should get the CV (matching CV item's `name.lower()` to `profile.lower()`), then storing that CV item's `content_profile_name` to `sttversion`, falling back to `profile` if CV item not found